### PR TITLE
fix(builder): watch for `pages/` creation when default page displayed

### PR DESCRIPTION
**Pull request by atinux**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Same as `store/` but applied for `pages/` creation.

## Checklist:

- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/builder/src/builder.js [hunk 1]
```diff
@@ -659,6 +659,10 @@ export default class Builder {
     if (this.ignore.ignoreFile) {
       nuxtRestartWatch.push(this.ignore.ignoreFile)
     }
+    // If default page displayed, watch for first page creation
+    if (this._nuxtPages && this._defaultPage) {
```
Comment (clarkdo) [inline packages/builder/src/builder.js]: Is `_defaultPage` is enough here ?

If `_defaultPage` is true, `_nuxtPages` must be `true`.

File: packages/cli/src/commands/dev.js [hunk 2]
```diff
@@ -48,7 +49,11 @@ export default {
     await nuxt.ready()
 
     // Start listening
-    await nuxt.server.listen()
+    await nuxt.server.listen(this._usedPort)
```
Comment (clarkdo) [inline packages/cli/src/commands/dev.js]: What if `server.port` in `nuxt.config` is changed at runtime ?

Comment (pi0) [reply packages/cli/src/commands/dev.js]: The `server.port` may be in use when we used the random port. checking for that value means we again fall back trying a used port and then choosing another random one. 

Alternatively, we can preserve a random chosen PORT in a global variable inside `listener` module and reuse it.

(PS: I think this fix can be moved to another PR for better history)

Comment (Atinux) [reply packages/cli/src/commands/dev.js]: Will create another PR for this.

File: packages/builder/src/builder.js (hunk 1 above)
Comment (Atinux) [reply packages/builder/src/builder.js]: Yes but readability is better :)

**Review by pi0 — APPROVED** [review]

File: packages/cli/src/commands/dev.js (hunk 2 above)
Comment (Atinux) [reply packages/cli/src/commands/dev.js]: Let's move to #5793 5793

**Pull request merged**